packages/integration-sdk-runtime/src/storage/FileSystemGraphObjectStore/__tests__/FileSystemGraphObjectStore.test.ts

```diff
@@ -426,6 +426,186 @@ describe('iterateRelationships', () => {
   });
 });
 
+describe('flush callbacks', () => {
+  test('#addEntity should call flush callback when buffer threshold reached', async () => {
+    const { storageDirectoryPath, store } = setupFileSystemObjectStore({
+      graphObjectBufferThreshold: 2,
+    });
+
+    let flushedEntitiesCollected: Entity[] = [];
+    let addEntitiesFlushCalledTimes = 0;
+
+    const e1 = generateEntity();
+    const e2 = generateEntity();
+    const e3 = generateEntity();
+
+    await store.addEntities(storageDirectoryPath, [e1], async (entities) => {
+      // Should not call first time because `graphObjectBufferThreshold` = 2
+      addEntitiesFlushCalledTimes++;
+      flushedEntitiesCollected = flushedEntitiesCollected.concat(entities);
+      return Promise.resolve();
+    });
+
+    await store.addEntities(
+      storageDirectoryPath,
+      [e2, e3],
+      async (entities) => {
+        // Should not call first time because `graphObjectBufferThreshold` = 2
+        addEntitiesFlushCalledTimes++;
+        flushedEntitiesCollected = flushedEntitiesCollected.concat(entities);
+        return Promise.resolve();
+      },
+    );
+
+    expect(addEntitiesFlushCalledTimes).toEqual(1);
+    expect(flushedEntitiesCollected).toEqual([e1, e2, e3]);
+  });
+
+  test('#addRelationships should call flush callback when buffer threshold reached', async () => {
+    const { storageDirectoryPath, store } = setupFileSystemObjectStore({
+      graphObjectBufferThreshold: 2,
+    });
+
+    let flushedRelationshipsCollected: Relationship[] = [];
+    let addRelationshipsFlushCalledTimes = 0;
+
+    const r1 = generateRelationship();
+    const r2 = generateRelationship();
+    const r3 = generateRelationship();
+
+    await store.addRelationships(
+      storageDirectoryPath,
+      [r1],
+      async (relationships) => {
+        // Should not call first time because `graphObjectBufferThreshold` = 2
+        addRelationshipsFlushCalledTimes++;
+        flushedRelationshipsCollected = flushedRelationshipsCollected.concat(
+          relationships,
+        );
+        return Promise.resolve();
+      },
+    );
+
+    await store.addRelationships(
+      storageDirectoryPath,
+      [r2, r3],
+      async (relationships) => {
+        // Should not call first time because `graphObjectBufferThreshold` = 2
+        addRelationshipsFlushCalledTimes++;
+        flushedRelationshipsCollected = flushedRelationshipsCollected.concat(
+          relationships,
+        );
+        return Promise.resolve();
+      },
+    );
+
+    expect(addRelationshipsFlushCalledTimes).toEqual(1);
+    expect(flushedRelationshipsCollected).toEqual([r1, r2, r3]);
+  });
+
+  test('#flushEntitiesToDisk should call flush callback when buffer threshold reached', async () => {
```

Comment:
Seems like copy-paste error? The buffer threshold has not been reached here.

Reply:
Thanks. Will fix.